src/App.tsx

```diff
@@ -1,39 +1,55 @@
-import React, {useEffect, useState} from 'react';
-import './App.css'
-import TodoList from "./TodoList.tsx";
-import AddTodoForm from "./AddTodoForm.tsx";
-import {Todo} from "./types.ts";
-
-const useSemiPersistentState = ():  [Todo[], React.Dispatch<React.SetStateAction<Todo[]>>] => {
-    const [todoList, setTodoList] = useState<Todo[]>(() => {
-        const savedTodoList = localStorage.getItem('savedTodoList');
-        return savedTodoList ? JSON.parse(savedTodoList) : [];
-    });
-    useEffect(()=> {
-        localStorage.setItem('savedTodoList', JSON.stringify(todoList));
-    }, [todoList])
-    return [todoList, setTodoList];
-}
+import React, { useEffect, useState } from 'react';
+import './App.css';
+import TodoList from './TodoList.tsx';
+import AddTodoForm from './AddTodoForm.tsx';
+import { Todo } from './types.ts';
 
 function App() {
-    const [todoList, setTodoList] = useSemiPersistentState();
+
+    const [todoList, setTodoList] = useState<Todo[]>([]);
+    const [isLoading, setIsLoading] = useState<boolean>(true);
+
+    const simulateLoading = (): Promise<{ data: { todoList: Todo[] } }> => {
+        return new Promise((resolve) => {
+            setTimeout(() => {
+                const savedTodoList = localStorage.getItem('savedTodoList');
+                resolve({
+                    data: { todoList: savedTodoList ? JSON.parse(savedTodoList) : [] },
+                });
+            }, 2000);
+        });
+    };
+
+    useEffect(() => {
+        simulateLoading().then((result) => {
```

Comment:
nit: looks good. as you get to do network requests, you'll want to add catch statements